### ISSUE TITLE
Bug: Game crashes when singular mod is enabled despite lack of changes between previous launches.

### ISSUE THREAD

**Issue by BonQerss** [Reloaded-Project/Reloaded-II#363]
**Describe the Bug**
Game crashing due to singular mod despite lack of changes between testing.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to (https://github.com/BonQerss/p3ppc.expandedsubmenu)
2. Click on Code and Download Zip
3. Build the mod in Visual Studio
4. Launch game with mod enabled in Reloaded

**Expected Behaviour**
Game launching with no issues and able to be tested.

**Screenshots**
[2024-04-18 17.44.28 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15028557/2024-04-18.17.44.28.P3P.txt)

**Additional Context**
Had this happen with another mod that I hadn't even edited on this new computer / hadn't touched in over 3 weeks.

**Comment by BonQerss** [comment]
Cache has been cleared 6 times, I have restarted my computer, I have rebuilt the DLL contained inside 4 times, I have rolled back all changes to back when it was working, I have redeployed ASI loader and launched from Steam, Nothing has fixed it.

https://youtu.be/tEQ46K1VBDM Unlisted video showing what happens as well as the two logs from when I launched it.
[2024-04-18 17.51.21 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15028659/2024-04-18.17.51.21.P3P.txt)
[2024-04-18 17.51.31 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15028662/2024-04-18.17.51.31.P3P.txt)

**Comment by Sewer56** [comment]
Log says

```
[10:44:28] [Reloaded] [LoaderAPI] Warning: GetController<IReloadedHooks> returned null.
```

Your install of `Reloaded.SharedLib.Hooks` is booked. Either due to AV, or due to a bad download.

**Comment by BonQerss** [comment]
Redownloaded it and it did not fix it.....,.

**Comment by Sewer56** [comment]
Throw me a new log after redownload

**Comment by BonQerss** [comment]
[2024-04-18 19.02.42 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15029428/2024-04-18.19.02.42.P3P.txt)

**Comment by Sewer56** [comment]
According to log, still borked

```
[12:02:43] [Reloaded] Loading: BF Emulator for File Emulation Framework
[12:02:43] [Reloaded] - AppId   : reloaded.universal.fileemulationframework.bf
[12:02:43] [Reloaded] - Location: C:\Users\airin\Downloads\Release\Mods\reloaded.universal.fileemulationframework.bf\ModConfig.json
[12:02:43] [Reloaded] [LoaderAPI] Warning: GetController<IReloadedHooks> returned null.
```

Alternatively, it could happen if hooks were removed as a direct or transitive dependency of your mods, although unlikely.

**Comment by BonQerss** [comment]
maybe i just redownloaded wrong. i just need to delete the old one and then launch and it should redownload, yeah?

**Comment by Sewer56** [comment]
Yeah. That usually works.

**Comment by BonQerss** [comment]
still crashes.
[2024-04-18 20.03.50 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15030021/2024-04-18.20.03.50.P3P.txt)

just wanna mention that that error has been present in pretty much all of my logs, going back to before this issue.
[2024-04-04 15.13.52 ~ P3P.txt](https://github.com/Reloaded-Project/Reloaded-II/files/15030030/2024-04-04.15.13.52.P3P.txt)

**Comment by Sewer56** [comment]
Hmm, how strange. And sort of a miracle stuff worked before.

I'll delegate this to @AnimatedSwine37 (so sorry for doing this). I'm still setting up my desk and don't have a working PC right now.

**Comment by AnimatedSwine37** [comment]
No worries I'll try and have a look tonight or on the weekend.

**Comment by AnimatedSwine37** [comment]
We investigated this in [a post in the Persona Modding discord server](https://discord.com/channels/746211612981198989/1231058598033883156) and found out the problem is unrelated to the missing hooks controller and was a bug with PAK Emulator.

I've made a [pr](https://github.com/Sewer56/FileEmulationFramework/pull/17) over on the File Emulation Framework repo with a fix for that.

The seemingly random nature of the crashes was due to them being caused by accessing allocated memory. If you were lucky the memory the file is loaded into could already be allocated so you wouldn't crash.
